### PR TITLE
refactor(database): cleanup database

### PR DESCRIPTION
**Pull request by ulfgebhardt**
## 🍰 Pullrequest

Cleanup database, reduce dependencies

### Issues

- None

### Todo

- [X] None

### CODE REVIEW

**Review by Mogge — COMMENTED** [review]
What about line 10 in `backend/src/server/createServer.ts` ?

**Review by Mogge — COMMENTED** [review]
What about the federation unit tests?

**Comment by clauspeterhuebner** [comment]
I do not understand the motivation behind this cleanup nor can i follow the changes. Is it possible to give an introduction or at least a more detailed description in the PR?

**Comment by ulfgebhardt** [comment]
> What about line 10 in `backend/src/server/createServer.ts` ?

You are talking about `import connection from '@/typeorm/connection'`?

This is part of the backend. The local(backend) connection is created with the local config values.

The typeorm include `import { Connection } from '@dbTools/typeorm'` cannot be removed - i tested this - the connection is bound to the path of the typeorm, therefore we need to use the same typeorm so the models (database) and the connection (backend) stem from the same typeorm source.

**Comment by ulfgebhardt** [comment]
> What about the federation unit tests?

They run, they have no dependencies on the stuff I changed

**Comment by ulfgebhardt** [comment]
> I do not understand the motivation behind this cleanup nor can i follow the changes. Is it possible to give an introduction or at least a more detailed description in the PR?

We figured out that the config values from backend are used for the database when including the database in the corresponding other project. This is confusing and considered `wrong`. Therefore I wanted to parameterized those parameters and figured out that we don't need all this stuff. E.g. we removed seeding form the database since we noticed we need the business logic.

**Review by clauspeterhuebner — COMMENTED** [review]
Even with your comments about the motivation of this PR i'am not able to evaluate the changes, because these are too much deep technology dependencies for me. Sorry, please wait for an approvement from @Mogge...

**Review by Elweyn — CHANGES_REQUESTED** [review]
I tested: 
- ``yarn dev_down``=> Works
- ``yarn dev_down``=> Works
- ``yarn reset``=> Works
- ``yarn dev_reset``=> Error:
```
hannesheine@Hanness-MacBook-Pro:~/workspace/gradido/database (cleanup-database$=) % yarn dev_reset
yarn run v1.22.19
$ cross-env TZ=UTC ts-node src/index.ts reset
Error: Script version and filename mismatch! version: 63 -> file system: 0063-event_link_fields.ts | database: 0063-event_link_fields.js
    at Migration.<anonymous> (/Users/hannesheine/workspace/gradido/database/node_modules/ts-mysql-migrate/src/migrations.ts:230:17)
    at Generator.next (<anonymous>)
    at fulfilled (/Users/hannesheine/workspace/gradido/database/node_modules/ts-mysql-migrate/dist/migrations.js:5:58)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
error Command failed with exit code 1.
```

**Review by Elweyn — APPROVED** [review]

File: database/src/prepare.ts [hunk 1]
```diff
@@ -1,15 +1,8 @@
...
- * This applies to all Database Operations like
- * creating, deleting, renaming Databases.
- */
+import { createConnection } from 'mysql2/promise'
 
-import { createConnection, RowDataPacket } from 'mysql2/promise'
 import CONFIG from './config'
 
-export default async (): Promise<void> => {
+const createDatabase = async (): Promise<void> => {
```
[the remark this answers is not in the document]
Comment (ulfgebhardt) [reply database/src/prepare.ts]: fixed

File: database/src/prepare.ts [hunk 2]
```diff
@@ -37,6 +32,9 @@ export default async (): Promise<void> => {
       console.log('Found and dropped old migrations table')
     }
   }
+  */
 
   await con.end()
 }
+
+export { createDatabase }
```
[the remark this answers is not in the document]
Comment (ulfgebhardt) [reply database/src/prepare.ts]: fixed

**Comment by ulfgebhardt** [comment]
> Member

The error occurred since you mixed dev and production datasets. Your database must be either migration.ts or migration.js. That always has been and is a limitation of our migration library.

**Review by Elweyn — APPROVED** [review]

**Pull request merged**